### PR TITLE
Delete dead PAL functions

### LINKED ISSUE #13503: Remove all dead PAL functions
There are likely still some PAL functions that are not used anymore. This issue tracks work needed to eliminate them.
There are two ways how PAL functions are used. First, coreclr native shared libraries and applications link to libcoreclrpal.a static library. And second, System.Private.CoreLib.dll PInvokes some of the PAL functions that are exposed by libcoreclr.so.
The functions that the native shared libraries and applications link to are listed in https://gist.github.com/janvorli/527f407a54c45e937b322ae373c00779.
The functions exposed by libcoreclr.so for PInvoke purposes include some of these and may add others. The list of the exposed functions is in https://github.com/dotnet/coreclr/blob/master/src/dlls/mscoree/mscorwks_unixexports.src.
The goal is to walk through https://github.com/dotnet/coreclr/blob/master/src/pal/inc/pal.h, find functions that are not in one of these two lists, verify that they are not needed by the PAL itself and then remove their implementation from PAL and their declarations from pal.h.

### PR DESCRIPTION
**Pull request by justinvp**
I think I got most of it.

Fixes #13503

### CODE REVIEW

**Comment by justinvp** [comment]
@dotnet-bot test Windows_NT x64 Formatting please (socket.error: [Errno 10054] An existing connection was forcibly closed by the remote host)

**Comment by justinvp** [comment]
cc: @janvorli, @danmosemsft

File: src/pal/src/file/file.cpp [hunk 1]
```diff
@@ -4319,261 +4119,6 @@ CreatePipe(
...
-        );
-
-    if (NO_ERROR != palError)
-    {
-        goto InternalLockFileExit;
-    }
-    
-    if (NULL != pLocalData->pLockController)
-    {
-        palError = pLocalData->pLockController->CreateFileLock(
```
Comment (jkotas) [inline src/pal/src/file/file.cpp]: Can the LockController be deleted as well?

Comment (justinvp) [reply src/pal/src/file/file.cpp]: I believe so... I was going to follow-up with another PR.

**Comment by danmosemsft** [comment]
Thanks @justinvp this is good cleanup ,appreciate it.

**Pull request merged**